app/javascript/components/patient/laboratory/FirstPositiveLab.js

```diff
@@ -0,0 +1,88 @@
+import React from 'react';
+import { PropTypes } from 'prop-types';
+import { Button } from 'react-bootstrap';
+import moment from 'moment';
+
+import LaboratoryModal from './LaboratoryModal';
+
+class FirstPositiveLab extends React.Component {
+  constructor(props) {
+    super(props);
+    this.state = {
+      showLabModal: false,
+    };
+  }
+
+  render() {
+    return (
+      <div>
+        {this.props.lab ? (
+          <div className={this.props.displayedLabClass || 'my-2'}>
+            <div className="first-positive-lab-result-header">
+              <div className="first-positive-lab-result-title">FIRST POSITIVE LAB RESULT</div>
+              <div className="edit-link">
+                <Button variant="link" id="edit-first_positive_lab" className="py-0" onClick={() => this.setState({ showLabModal: true })}>
+                  Edit
+                </Button>
+              </div>
+              <div className="edit-link">
+                <Button variant="link" id="delete-first_positive_lab" className="py-0" onClick={() => this.props.onChange(null)}>
+                  Delete
+                </Button>
+              </div>
+            </div>
+            <div>
+              <span className="first-positive-lab-result-field-name">Type: </span>
+              <span>{this.props.lab.lab_type || '--'}</span>
+            </div>
+            <div>
+              <span className="first-positive-lab-result-field-name">Specimen Collection Date: </span>
+              <span>{this.props.lab.specimen_collection ? moment(this.props.lab.specimen_collection, 'YYYY-MM-DD').format('MM/DD/YYYY') : '--'}</span>
+            </div>
+            <div>
+              <span className="first-positive-lab-result-field-name">Report Date: </span>
+              <span>{this.props.lab.report ? moment(this.props.lab.report, 'YYYY-MM-DD').format('MM/DD/YYYY') : '--'}</span>
+            </div>
+            <div>
+              <span className="first-positive-lab-result-field-name">Result: </span>
+              <span>{this.props.lab.result || '--'}</span>
+            </div>
+          </div>
+        ) : (
+          <div className="mb-2">
+            <div className="pb-2">
+              <span>POSITIVE LAB RESULT</span>
+            </div>
+            <Button variant="primary" size={this.props.size || 'md'} onClick={() => this.setState({ showLabModal: true })}>
+              <i className="fas fa-plus-square mr-2"></i>
+              Enter Lab Result
+            </Button>
+          </div>
+        )}
+        {this.state.showLabModal && (
+          <LaboratoryModal
+            lab={this.props.lab}
+            specimenCollectionRequired={true}
+            onlyPositiveResult={true}
+            submit={lab => {
+              this.setState({ showLabModal: false });
+              this.props.onChange(lab);
```

Comment:
can we move this to the set state callback?

Reply:
is there a particular reason for this? there's no need/reason to wait for the modal to close before calling `onChange`. if anything, leaving it out of the callback would make it more performant because it doesn't have to wait/rely on `setState`

Reply:
https://github.com/SaraAlert/SaraAlert/pull/861/files#diff-8147b18fdde2f3d544f4ab613568fc11952197eb2907e5d2a55ae37756aea1a2R69